addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/TestRemoteFileIncludeUnitTest.java

```diff
@@ -19,12 +19,53 @@
  */
 package org.zaproxy.zap.extension.ascanrules;
 
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.zaproxy.zap.extension.ascanrules.utils.Constants.NULL_BYTE_CHARACTER;
+
+import fi.iki.elonen.NanoHTTPD.IHTTPSession;
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Plugin.AttackStrength;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
 /** Unit test for {@link TestRemoteFileInclude}. */
 public class TestRemoteFileIncludeUnitTest
         extends ActiveScannerAppParamTest<TestRemoteFileInclude> {
 
     @Override
     protected TestRemoteFileInclude createScanner() {
-        return new TestRemoteFileInclude();
+        TestRemoteFileInclude scanner = new TestRemoteFileInclude();
+        scanner.setConfig(new ZapXmlConfiguration());
+        return scanner;
+    }
+
+    @Test
+    public void shouldRaiseAlertIfResponseHasRemoteFileContentAndPayloadIsNullByteBased()
+            throws HttpMalformedHeaderException {
+        // Given
+        NullByteVulnerableServerHandler vulnServerHandler =
+                new NullByteVulnerableServerHandler("/", "p", Tech.Linux) {
+                    protected String getContent(IHTTPSession session) {
+                        String value = getFirstParamValue(session, "p");
+                        if (value.contains(NULL_BYTE_CHARACTER)) {
+                            return "<!doctype html><html itemscope=\"\" itemtype=\"http://schema.org/WebPage\" "
```

Comment:
This can be simplified `return "<html><title>Google</title></html>";`.

Reply:
hmm. will do that.